iml-agent/src/action_plugins/stratagem/action_filesync.rs

```diff
@@ -5,33 +5,54 @@
 use crate::{agent_error::ImlAgentError, env, lustre::search_rootpath};
 use futures::{future::try_join_all, future::Future, stream, StreamExt, TryStreamExt};
 use iml_wire_types::{FidError, FidItem};
-use liblustreapi::LlapiFid;
 use std::path::PathBuf;
 use structopt::clap::arg_enum;
 use tokio::{fs, process::Command};
 
+arg_enum! {
+    #[derive(Debug, serde::Deserialize, Clone, Copy, PartialEq)]
+    pub enum ActionType {
+    Push,
+    Pull,
+    }
+
+}
 struct Work {
+    file_path: PathBuf,
     fid: FidItem,
-    src_file: String,
-    dest_file: String,
 }
 
 const LARGEFILE: u64 = 1024 * 1024 * 1024;
 
 fn do_rsync<'a>(
+    src_root: &'a PathBuf,
+    dest_root: &'a PathBuf,
     work_list: &'a [Work],
```

Comment:
```suggestion
    work_list: Vec<Work>,
```

This would be better to pass by value instead of reference. Once this fn owns the `Vec` it can return the fid data directly instead of cloning it.

Reply:
Done